GLTFSDK.Samples/Deserialize/Source/main.cpp

```diff
@@ -0,0 +1,304 @@
+        std::cout << "Asset Generator:  " << document.asset.generator << "\n";
+        std::cout << "Asset Copyright:  " << document.asset.copyright << "\n\n";
+
+        // Scene Info
+        std::cout << "Scene Count: " << document.scenes.Size() << "\n";
+
+        if (document.scenes.Size() > 0U)
+        {
+            std::cout << "Default Scene Index: " << document.GetDefaultScene().id << "\n\n";
+        }
+        else
+        {
+            std::cout << "\n";
+        }
+
+        // Entity Info
+        std::cout << "Node Count:     " << document.nodes.Size() << "\n";
+        std::cout << "Camera Count:   " << document.cameras.Size() << "\n";
+        std::cout << "Material Count: " << document.materials.Size() << "\n\n";
+
+        // Mesh Info
+        std::cout << "Mesh Count: " << document.meshes.Size() << "\n";
+        std::cout << "Skin Count: " << document.skins.Size() << "\n\n";
+
+        // Texture Info
+        std::cout << "Image Count:   " << document.images.Size() << "\n";
+        std::cout << "Texture Count: " << document.textures.Size() << "\n";
+        std::cout << "Sampler Count: " << document.samplers.Size() << "\n\n";
+
+        // Buffer Info
+        std::cout << "Buffer Count:     " << document.buffers.Size() << "\n";
+        std::cout << "BufferView Count: " << document.bufferViews.Size() << "\n";
+        std::cout << "Accessor Count:   " << document.accessors.Size() << "\n\n";
+
+        // Animation Info
+        std::cout << "Animation Count: " << document.animations.Size() << "\n\n";
+
+        for (const auto& extension : document.extensionsUsed)
+        {
+            std::cout << "Extension Used: " << extension << "\n";
+        }
+
+        if (!document.extensionsUsed.empty())
+        {
+            std::cout << "\n";
+        }
+
+        for (const auto& extension : document.extensionsRequired)
+        {
+            std::cout << "Extension Required: " << extension << "\n";
+        }
+
+        if (!document.extensionsRequired.empty())
+        {
+            std::cout << "\n";
+        }
+    }
+
+    // Uses the Document and GLTFResourceReader classes to print information about various glTF binary resources
+    void PrintResourceInfo(const Document& document, const GLTFResourceReader& resourceReader)
+    {
+        // Use the resource reader to get each mesh primitive's position data
+        for (const auto& mesh : document.meshes.Elements())
+        {
+            std::cout << "Mesh: " << mesh.id << "\n";
+
+            for (const auto& meshPrimitive : mesh.primitives)
+            {
+                std::string accessorId;
+
+                if (meshPrimitive.TryGetAttributeAccessorId(ACCESSOR_POSITION, accessorId))
+                {
+                    const Accessor& accessor = document.accessors.Get(accessorId);
+
+                    const auto data = resourceReader.ReadBinaryData<float>(document, accessor);
```

Comment:
Might be nice to print this data. A lot of lines of code in here used to print trivial info which you can see just by inspecting the JSON, whereas this would be something that is not easy to do without the SDK.

Reply:
Can use our Export sample to test this and not have it print a ridiculous number of positions.

Reply:
I've not going to change this. The sample demonstrates how to get the position data as a vector. I'm pretty sure other devs can figure out what to do with it after that!